File: frontend/src/types/actions/token.ts
import { z } from "zod";

export const tokenGetRequestSchema = z.object({
  api_key: z.string(),
  table_name: z.string(),
});

export type TokenGetRequest = z.infer<typeof tokenGetRequestSchema>;

export const tokenGetResponseSchema = z.object({
  is_authenticated: z.boolean(),
});

export type TokenGetResponse = z.infer<typeof tokenGetResponseSchema>;

File: backend/app/models/token.py
from typing import Optional

from pydantic import BaseModel


class TokenPostRequest(BaseModel):
    api_key: str
    access_token: str
    refresh_token: Optional[str]
    client_id: str
    client_secret: str
    table_name: str


class TokenGetRequest(BaseModel):
    api_key: str
    table_name: str


class TokenGetResponse(BaseModel):
    is_authenticated: bool
